benchmarks/benchmarks/generate_data.py

```diff
@@ -0,0 +1,127 @@
+"""
+Scripts for generating supporting data for benchmarking.
+
+Data generated using iris-esmf-regrid should use
+:func:`run_function_elsewhere`, which means that data is generated using a
+fixed version of iris-esmf-regrid and a fixed environment, rather than those
+that get changed when the benchmarking run checks out a new commit.
+
+Downstream use of data generated 'elsewhere' requires saving; usually in a
+NetCDF file. Could also use pickling but there is a potential risk if the
+benchmark sequence runs over two different Python versions.
+
+"""
+from inspect import getsource
+from subprocess import CalledProcessError, check_output, run
+from os import environ
+from pathlib import Path
+from textwrap import dedent
+
+from iris import load_cube
+
+#: Python executable used by :func:`run_function_elsewhere`, set via env
+#:  variable of same name. Must be path of Python within an environment that
+#:  supports iris-esmf-regrid and has iris-esmf-regrid installed via
+#:  ``pip install -e``.
+DATA_GEN_PYTHON = environ.get("DATA_GEN_PYTHON", "")
```

Comment:
Would a more sensible default value be `"python"`? otherwise suggest a normal dict lookup and have the code fail here rather than the check output line with `["","-c","a = True"]` below.

Reply:
Using the default `python` will lead to the unwanted behaviour that we are trying to avoid here - that the generated synthetic data may change with each commit.

I wanted to capture two things here - the user providing no value at all, or providing an invalid value. Perhaps this has ended up a bit hacky by me trying to encapsulate everything within one try-except. I could certainly have a catch for no value at all, and then something else for an invalid path.

Reply:
b9de1fcc580e2df822231ca637513bd9da839ca5